libraries/oneview_resource_base.rb

```diff
@@ -31,7 +31,7 @@ module ResourceBase
     def create_or_update(item = nil, method_1 = :create, method_2 = :update)
       ret_val = false
       item ||= load_resource
-      temp = item.data.clone
+      temp = Marshal.load(Marshal.dump(item.data))
```

Comment:
For nested data hashes, it would override the copy when we did a retrieve. The clone method is just a shallow copy, so if you play with how it works on nested hashes, second-level hashes or greater are just pointers to the original. The retrieve will overwrite that copy, causing the like? method to say they're the same, when they might not be.